File: CompleteSolutions/khabe_poopak.cpp
// https://quera.org/problemset/15124/
// writed with chatGPT python to cpp

#include <iostream>
#include <vector>
using namespace std;

vector<int> divisors(int n) {
    vector<int> res;
    for(int i = 1; i <= n; i++) {
        if(n % i == 0) {
            res.push_back(i);
        }
    }
    return res;
}

int count_pairs(vector<int> list1, vector<int> list2, int n) {
    int count = 0;
    for(int i = 0; i < list1.size(); i++) {
        for(int j = 0; j < list2.size(); j++) {
            if(list1[i] + list2[j] <= n) {
                count++;
            }
        }
    }
    return count;
}

int main() {
    int a, b, x;
    cin >> a >> b >> x;
    vector<int> div1 = divisors(a);
    vector<int> div2 = divisors(b);
    cout << count_pairs(div1, div2, x) << endl;
    return 0;
}

File: CompleteSolutions/khabe_poopak.py
# https://quera.org/problemset/15124/

a, b, x = list(map(int, input().split()))

# functions write with chatGPT help

# def divisors(n):
#     result = []
#     for i in range(1, n+1):
#         if n % i == 0:
#             result.append(i)
#     return result

# def count_pairs(list1, list2, n):
#     count = 0
#     for x in list1:
#         for y in list2:
#             if x + y <= n:
#                 count += 1
#     return count

def divisors(n):
    return [i for i in range(1, n+1) if n % i == 0]

def count_pairs(list1, list2, n):
    return sum([1 for x in list1 for y in list2 if x + y <= n])

print(count_pairs(divisors(a), divisors(b), x))